src/App.tsx

```diff
@@ -1,223 +1,110 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
+import { TodosSelection } from './components/TodosSelection/TodosSelection';
+import { TodoForm } from './components/TodoForm/TodoForm';
+import { TodoList } from './components/TodoList/TodoList';
+
+import { Todo } from './types/Todo';
+import { TodosFilter } from './types/TodosFilter';
+import { getTodos } from './api/todos';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filtredTodos, setFiltredTodos] = useState<Todo[]>([]);
+  const [
+    statusToFilter,
+    setStatusToFilter,
+  ] = useState<TodosFilter>(TodosFilter.All);
+  const [hasError, setHasError] = useState(false);
+
+  const closeNotification = useCallback(() => setHasError(false), []);
+
+  const filterTodosBy = useCallback((status: TodosFilter) => {
+    const filtredByStatusTodos = todos.filter(({ completed }) => {
+      switch (status) {
+        case TodosFilter.Active:
+          return !completed;
+
+        case TodosFilter.Completed:
+          return completed;
+
+        default:
+          return true;
+      }
+    });
+
+    setStatusToFilter(status);
+    setFiltredTodos(filtredByStatusTodos);
+  }, [todos]);
```

Comment:
State with filtered todos is not required here, we could save result of filtering to constant using useMemo to avoid unnecessary filtration. Let's do it this way

Reply:
Thx, it's really good point, and it's looks easier!